src/Config/Provider/AutodnsProvider.php

```diff
@@ -0,0 +1,1347 @@
+<?php
+
+namespace App\Config\Provider;
+
+use App\Entity\Domain;
+use Psr\Cache\CacheItemInterface;
+use Symfony\Component\HttpClient\HttpOptions;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class AutodnsProvider extends AbstractProvider
+{
+    private const BASE_URL = 'https://api.autodns.com';
+
+    /**
+     * Order a domain name with the Gandi API.
+     *
+     * @throws \Exception
+     * @throws TransportExceptionInterface
+     * @throws DecodingExceptionInterface
+     */
+    public function orderDomain(Domain $domain, bool $dryRun = false): void
+    {
+        if (!$domain->getDeleted()) {
+            throw new \InvalidArgumentException('The domain name still appears in the WHOIS database');
+        }
+
+        $ldhName = $domain->getLdhName();
+        if (!$ldhName) {
+            throw new \InvalidArgumentException('Domain name cannot be null');
+        }
+
+        $authData = self::verifyAuthData($this->authData, $this->client);
+
+        $zoneCheck = $this->client->request(
+            'POST',
+            '/v1/zone/_search?keys=name',
+            (new HttpOptions())
+                ->setAuthBasic($authData["username"], $authData["password"])
+                ->setHeader('Accept', 'application/json')
+                ->setHeader('X-Domainrobot-Context', $authData["context"])
+
+                ->setBaseUri(self::BASE_URL)
+                ->setJson([
+                    'filters' => [
+                        [
+                            "key" => "name",
+                            "value" => $ldhName,
+                            "operator" => "EQUAL",
+                        ]
+                    ],
+                ])
+                ->toArray()
+        )->toArray();
+
+        if (empty($zoneCheck["data"])) {
+            // The domain not yet exists in DNS Server, we create them
+
```

Comment:
There are plans for a unified default zone across providers when a domain is purchased, unless autodns requires this call to be there, i think this should be delayed until the zone abstraction is thought about. What do you think @maelgangloff

Reply:
Same opinion. We want to buy the domain name as quickly as possible. The definition of the zone can be done later by human action.
Unless it is a prerequisite for the purchase of the domain name, let's not waste time setting up a DNS zone.

Reply:
I agree. And I did a quick test, because last time (many years ago) I tried to register a domain without adding the zone before, resulted in an error from DENIC. 
(They require to have at least 2 Nameserver which answer with a valid response for the domain ([Docs](https://nast.denic.de/docs/spec/spec_en.html))

_Short term result of this test:_ The domain was registered at the moment, but I got bad warnings from DENIC in registration protocol:
```
Nameserver error
ERROR: 133 Answer must be authoritative (nameserver, ip, proto, record) ([c.ns14.net](http://c.ns14.net/), 2a01:130:2000:118:89:146:248:231, udp, NS)
53300102912
Nameserver error
ERROR: 901 Unexpected RCODE (nameserver, ip, proto, record) ([c.ns14.net](http://c.ns14.net/), 2a01:130:2000:118:89:146:248:231, udp, NS)
53300102912
```

But I suppose I can register domain and handle DNS errors afterwards, because I get some time to correct from DENIC. 

So I will remove the DNS topic of Domains at the moment. 
But to have a unified DNS definition, would be helpfull, because this can be directly executed after orderDomain automatically instead of manually.